### PR TITLE
package/libspatialite: add conflict, new version

### PR DESCRIPTION
**Pull request by Sinan81**
- older libspatialite fails to build with newer geos@3.10 onwards
- add version 5.1.0

### CODE REVIEW

**Comment by Sinan81** [comment]
depends on https://github.com/spack/spack/pull/42574

**Comment by Sinan81** [comment]
pinging @adamjstewart since this is GIS stuff.

File: var/spack/repos/builtin/packages/libspatialite/package.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Libspatialite(AutotoolsPackage):
 
     license("MPL-1.1")
 
+    version("5.1.0", sha256="43be2dd349daffe016dd1400c5d11285828c22fea35ca5109f21f3ed50605080")
     version("5.0.1", sha256="eecbc94311c78012d059ebc0fae86ea5ef6eecb13303e6e82b3753c1b3409e98")
     version("5.0.0", sha256="7b7fd70243f5a0b175696d87c46dde0ace030eacc27f39241c24bac5dfac6dac")
     # Must download manually from:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libspatialite/package.py]: Any reason we shouldn't deprecate all of these weird/alpha/beta versions? I don't see any packages that require them.

File: var/spack/repos/builtin/packages/libspatialite/package.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ class Libspatialite(AutotoolsPackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libspatialite/package.py:15]: manual_download is only true for one version, correct? I would remove this

File: var/spack/repos/builtin/packages/libspatialite/package.py (hunk 1 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/libspatialite/package.py]: fixed! 4.3.0a was released after 4.3.0. so let's keep it.

File: var/spack/repos/builtin/packages/libspatialite/package.py [hunk 3]
```diff
@@ -18,33 +16,22 @@ class Libspatialite(AutotoolsPackage):
...
-    #     https://github.com/spack/spack/issues/2489
-    version(
-        "5.0.0.2.c7f67038bf",
-        sha256="f8100f71b769c7db066c6f938af6b00e920e4b90ac14c00a4f3ed7171565caab",
-        url="file://%s/SpatiaLite-c7f67038bf.tar.gz" % os.getcwd(),
-    )
-
-    version(
-        "5.0.0-beta0", sha256="caacf5378a5cfab9b8e98bb361e2b592e714e21f5c152b795df80d0ab1da1c42"
-    )
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libspatialite/package.py:35]: Versions should be deprecated before removing them: https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

File: var/spack/repos/builtin/packages/libspatialite/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/libspatialite/package.py]: Could still deprecate those ancient versions if we don't want to support them, up to you

Comment (Sinan81) [reply var/spack/repos/builtin/packages/libspatialite/package.py]: version 5.0.0.2.c7f67038bf isn't available on downloads. So, I'll remove it without deprecate notice.

**Comment by Sinan81** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/libspatialite/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/libspatialite/package.py:15]: Still need to remove `manual_download`

Comment (Sinan81) [reply var/spack/repos/builtin/packages/libspatialite/package.py:15]: oops. totally missed it. fixed now.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**